src/main/java/cloud/eppo/EppoClient.java

```diff
@@ -0,0 +1,477 @@
+package cloud.eppo;
+
+import static cloud.eppo.Utils.getMD5Hex;
+import static cloud.eppo.Utils.throwIfEmptyOrNull;
+
+import cloud.eppo.logging.Assignment;
+import cloud.eppo.logging.AssignmentLogger;
+import cloud.eppo.ufc.dto.EppoValue;
+import cloud.eppo.ufc.dto.FlagConfig;
+import cloud.eppo.ufc.dto.SubjectAttributes;
+import cloud.eppo.ufc.dto.VariationType;
+import java.util.HashMap;
+import java.util.Map;
+
+import cloud.eppo.ufc.dto.adapters.EppoModule;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EppoClient {
+  private static final Logger log = LoggerFactory.getLogger(EppoClient.class);
+  private final ObjectMapper mapper = new ObjectMapper().registerModule(EppoModule.eppoModule()); // TODO: is this the best place for this?
+  private static final String DEFAULT_HOST = "https://fscdn.eppo.cloud";
+  private static final boolean DEFAULT_IS_GRACEFUL_MODE = true;
+
+  private final ConfigurationRequestor requestor;
+  private final AssignmentLogger assignmentLogger;
+  private final boolean isGracefulMode;
+  private final String sdkName;
+  private final String sdkVersion;
+  private final boolean isConfigObfuscated;
+
+  private static EppoClient instance;
+
+  // Fields useful for testing in situations where we want to mock the http client or configuration
+  // store (accessed via reflection)
+  /**
+   * @noinspection FieldMayBeFinal
+   */
+  private static EppoHttpClient httpClientOverride = null;
+
+  /**
+   * @noinspection FieldMayBeFinal
+   */
+  private static ConfigurationStore configurationStoreOverride = null;
+
+  private EppoClient(
+      String apiKey,
+      String sdkName,
+      String sdkVersion,
```

Comment:
SDK name and version are passed in via the constructor/builders to be used by the upstream Java and Android SDKs